From ea529e2f1c6108d5d36e965f4f4c5c3c5b38944b Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Sat, 30 Nov 2019 22:53:30 +0530
Subject: [PATCH 416/767] octeontx2-pf: Process CQE_TX first and then CQE_RX

commit 02237bdd58d65c17d5b502bb027290dd7a281bf2 from
git@git.assembla.com:cavium/WindRiver.linux.git

Process transmitted pkt's CQEs first so that the freed skbs
are recycled for received packets.

Change-Id: Ia399659851c4b8181619ee212d1c9ee26711efc6
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/19652
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
index 6c1a1ef96fbc..6f8da55f60a5 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.c
@@ -470,7 +470,7 @@ int otx2_napi_handler(struct napi_struct *napi, int budget)
 	pfvf = (struct otx2_nic *)cq_poll->dev;
 	qset = &pfvf->qset;
 
-	for (i = 0; i < CQS_PER_CINT; i++) {
+	for (i = CQS_PER_CINT - 1; i >= 0; i--) {
 		cq_idx = cq_poll->cq_ids[i];
 		if (cq_idx == CINT_INVALID_CQ)
 			continue;
-- 
2.31.1

